### PR TITLE
chore: Convert existing CI workflows to maven

### PR DESCRIPTION
**Pull request by lqiu96**
This PR is to convert existing CI workflows to maven. There are a few changes in this PR:
- Remove the old, unused CIs
- Replace most of the jobs with Github Actions
- Add linting, formatting, and checkstyles
- Set the required CIs via sync-repo-settings.json

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -5,53 +5,59 @@ branchProtectionRules:
   - pattern: main
     isAdminEnforced: true
     requiredStatusCheckContexts:
-      - units
-      - format
+      - units (8)
+      - units (11)
+      - units (17)
```
Comment (blakeli0) [inline .github/sync-repo-settings.yaml:10]: lint should also be required.

File: .kokoro/build.sh [hunk 2]
```diff
@@ -15,11 +15,70 @@
...
+      -Dclirr.skip=true \
+      -Denforcer.skip=true \
+      -Dcheckstyle.skip=true \
+      -Dflatten.skip=true \
+      -Danimal.sniffer.skip=true \
+      -Dmaven.wagon.http.retryHandler.count=5 \
+      test
+    RETURN_CODE=$?
+    ;;
+  clirr)
```
Comment (blakeli0) [inline .kokoro/build.sh:63]: I don't see a clirr check in ci, maybe we can add one as an optional?

File: build.gradle [hunk 3]
```diff
@@ -272,7 +272,7 @@ afterEvaluate {
               name = 'GoogleAPIs'
               email = 'googleapis@googlegroups.com'
               url = 'https://github.com/googleapis'
-              organization = 'Google, Inc.'
```
Comment (blakeli0) [inline build.gradle:275]: What's the reason for this change?

File: .github/sync-repo-settings.yaml (hunk 1 above)
Comment (lqiu96) [reply .github/sync-repo-settings.yaml:10]: Sure, adding.

File: java.header [hunk 4]
```diff
@@ -1,15 +1,29 @@
 ^/\*$
-^ \* Copyright \d\d\d\d,? Google (Inc\.|LLC)( All [rR]ights [rR]eserved\.)?$
+^ \* Copyright 20\d\d Google LLC$
```
Comment (blakeli0) [inline java.header:2]: What's the reason for the copyright change?

File: .github/sync-repo-settings.yaml [hunk 5]
```diff
@@ -5,53 +5,59 @@ branchProtectionRules:
...
     requiresCodeOwnerReviews: true
     requiresStrictStatusChecks: true
   - pattern: java7
     isAdminEnforced: true
     requiredStatusCheckContexts:
-      - units
-      - format
+      - units (8)
+      - units (11)
+      - units (17)
```
Comment (lqiu96) [inline .github/sync-repo-settings.yaml:30]: Searching for non cloud libraries dependent on java 7

File: .kokoro/build.sh [hunk 6]
```diff
@@ -15,11 +15,70 @@
...
+if [ "${REPORT_COVERAGE}" == "true" ]; then
+  bash ${KOKORO_GFILE_DIR}/codecov.sh
+fi
+
+# fix output location of logs
+bash .kokoro/coerce_logs.sh
+
+if [[ "${ENABLE_FLAKYBOT}" == "true" ]]; then
+  chmod +x ${KOKORO_GFILE_DIR}/linux_amd64/flakybot
+  ${KOKORO_GFILE_DIR}/linux_amd64/flakybot -repo=googleapis/google-cloud-java
```
Comment (lqiu96) [inline .kokoro/build.sh:80]: api-common-java

File: .kokoro/build.sh [hunk 7]
```diff
@@ -15,11 +15,70 @@
...
+fi
```
Comment (lqiu96) [inline .kokoro/build.sh:81]: Option: Delete this section?

File: .kokoro/build.sh (hunk 2 above)
Comment (lqiu96) [reply .kokoro/build.sh:63]: Adding in CLIRR check

File: java.header (hunk 4 above)
Comment (lqiu96) [reply java.header:2]: Reverting.

**Comment by lqiu96** [comment]
We are going to determine if the java.header file should have `Google, Inc.` vs `Google, LLC.`

File: java.header [hunk 8]
```diff
@@ -1,15 +1,29 @@
...
+^ \* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR$
+^ \* A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT$
+^ \* OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,$
+^ \* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES \(INCLUDING, BUT NOT$
+^ \* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,$
+^ \* DATA, OR PROFITS; OR BUSINESS INTERRUPTION\) HOWEVER CAUSED AND ON ANY$
+^ \* THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT$
+^ \* \(INCLUDING NEGLIGENCE OR OTHERWISE\) ARISING IN ANY WAY OUT OF THE USE$
+^ \* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.$
+^ \*/$
```
Comment (lqiu96) [inline java.header:29]: This is the same as `license-header-javadoc.txt` which gradle has been using. Nothing about the license changes except that maven uses this file. 

I've updated it to accept Google Inc and LLC

**Review by blakeli0 — APPROVED** [review]

**Comment by lqiu96** [comment]
I'm going to re-add in the old `ci/units` and `ci/format` jobs to make the status pass. I'll create a separate PR to remove them since they are required checks for now.
